addons/input-overlay/overlay-widget/overlay-widget.gd

```diff
@@ -1,39 +1,45 @@
 @tool
-extends Panel
 class_name OverlayWidget
+extends Panel
 
 @onready var mouse_overlay = %MouseOverlay
 
-var a = false
-var s = false
-var d = false
-var lmb = false
-var rmb = false
-var mmb = false
-var mmscroll = false
-
-
-# TODO: add support for multiple buttons pressed at the same time.
-func handle_events():
-	if lmb:
-		mouse_overlay.set_mouse_pressed(true, "lmb")
-	elif rmb:
-		mouse_overlay.set_mouse_pressed(true, "rmb")
-	elif mmb:
-		mouse_overlay.set_mouse_pressed(true, "mmb")
-	else:
-		mouse_overlay.set_mouse_pressed(false)
+func handle_mouse_events(event):
+	var buttons = _get_buttons(event)
+	mouse_overlay.update_mouse_overlay(buttons)
 
 
 func receive_input(event):
 	if event is InputEventMouseButton:
-		lmb = event.button_index == MOUSE_BUTTON_LEFT and event.pressed
-		rmb = event.button_index == MOUSE_BUTTON_RIGHT and event.pressed
-		mmb = (event.button_index in [MOUSE_BUTTON_MIDDLE, MOUSE_BUTTON_WHEEL_DOWN, MOUSE_BUTTON_WHEEL_UP] and event.pressed)
-#	if event is InputEventKey:
-#		if event.keycode == KEY_ESCAPE and event.pressed:
-#			disable() if visible == true else enable()
-	handle_events()
+		handle_mouse_events(event)
+
+
+func _get_buttons(event):
+	var buttons = {"lmb": false, "mmb": false, "rmb": false}
+	var button_mask = event.button_mask
+	var button_index = event.button_index
+	# Scroll events can mess with the state of the buttons
+	if (button_index == MOUSE_BUTTON_WHEEL_UP or button_index == MOUSE_BUTTON_WHEEL_DOWN):
+		return {}
+	if event.pressed:
+		# When some button is pressed, update the state of all buttons to deal
+		# with problems related to the popup menu
+		if button_mask & MOUSE_BUTTON_MASK_LEFT == MOUSE_BUTTON_MASK_LEFT:
+			buttons["lmb"] = true
+		if button_mask & MOUSE_BUTTON_MASK_MIDDLE == MOUSE_BUTTON_MASK_MIDDLE:
+			buttons["mmb"] = true
+		if button_mask & MOUSE_BUTTON_MASK_RIGHT == MOUSE_BUTTON_MASK_RIGHT:
+			buttons["rmb"] = true
+	else:
+		# For some reason, the event fired on button release only reports the
+		# released button
+		if button_index == MOUSE_BUTTON_LEFT:
+			buttons = { "lmb": false }
+		elif button_index == MOUSE_BUTTON_MIDDLE:
+			buttons = { "mmb": false }
+		elif button_index == MOUSE_BUTTON_RIGHT:
+			buttons = { "rmb": false }
+	return buttons
```

Comment:
Since this is related to the mouse I would move it in the `mouse-buttons.gd` script.

Reply:
Yeah, probably its better to put there, I'll fix on the next PR